osu.Game/Screens/Edit/Compose/HitObjectContainerEventQueue.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Objects.Drawables;
+using osu.Game.Rulesets.UI;
+
+namespace osu.Game.Screens.Edit.Compose
+{
+    /// <summary>
+    /// A queue which processes events from the many <see cref="HitObjectContainer"/>s in a nested <see cref="Playfield"/> hierarchy.
+    /// </summary>
+    internal class HitObjectContainerEventQueue : Component
```

Comment:
It is not a "queue" if the event ordering is not preserved.

Reply:
Is "buffer" more acceptable?

Reply:
I definitely prefer buffer.